lib/actions/deploy.js

```diff
@@ -263,16 +263,6 @@ Examples:
 		appName = appName_raw || appName || options.application;
 		delete options.application;
 
-		options.convertEol = options['convert-eol'] || false;
-		delete options['convert-eol'];
-		if (options.convertEol && !options.build) {
-			return Promise.reject(
-				new ExpectedError(
-					'The --eol-conversion flag is only valid with --build.',
```

Comment:
I've removed this check / requirement because I noticed that we also need EOL conversion (or prevention with `--noconvert-eol`) even if `--build` is not specified, because an image will be built even if `--build` is not specified in case the respective image is not found locally. In this case, I also realised that the EOL logic for `balena build` and `balena deploy` become identical, so I moved the flag from `opts.convertEol` to `composeOpts.convertEol` so that the logic was centralised in `compose.js` instead of duplicated in `build.js` and `deploy.js`.